### ISSUE TITLE
Can't run "npm pack" and "npm install tarball" like I used to in the past. (adding "base" word?)

### ISSUE THREAD

**Issue by quiringk** [valor-software/ngx-bootstrap#1358]
A few months ago I worked with @valorkin and added some functionality for "current slide" in the carousel. These changes were added in a branch but never added to the main repo (atleast, I don't think so.) So when new changes come out for previous versions, what I do is clone ng2-bootstrap, add the changes from that carousel branch, run npm pack, and then npm install "generated ng2-bootstrap tarball" inside of my main project (instead of using package.json like normal.) This has worked until today when I installed 1.1.16-7 to fix the modal application ref errors. The fix worked, however the npm pack generated a tarball like 

`ng2-bootstrap-base-1.1.16-7.tgz` with the word base in it?

When I run npm install I don't see any of the components inside ng2-bootstrap-base directory in node_modules. What is this new base word doing in there? 

Now I am getting this error 
`zone.js:1382 GET http://localhost:3000/node_modules/ng2-bootstrap/bundles/ng2-bootstrap.umd.js 404 (Not Found)`
which I'm not getting if I just install ng2-bootstrap 1.1.16-7 normally thru package.json (the .umd.js file is not in the ng2-bootstrap-base directory)

**Comment by valorkin** [comment]
Hey, @macroorganizm is working on your changes
he is covering it with tests to not break existing functionality
so I am expecting to publish them this week

https://github.com/valor-software/ng2-bootstrap/pull/1298

**Comment by valorkin** [comment]
@quiringk check dist folder
it has a package, and it is published from dist folder
may be running npm pack from there will help?

**Comment by quiringk** [comment]
@valorkin I dont see a dist folder after running npm install on the clone. I tried running npm start and got this typeerror:

```
ngCompiler.ReflectorHost is not a constructor
TypeError: ngCompiler.ReflectorHost is not a constructor
    at AotPlugin._setupOptions (C:\Projects\ng2-bootstrap-3\node_modules\@ngtools\webpack\src\plugin.js:136:31)
    at new AotPlugin (C:\Projects\ng2-bootstrap-3\node_modules\@ngtools\webpack\src\plugin.js:37:14)
    at Object.exports.getWebpackAotConfigPartial (C:\Projects\ng2-bootstrap-3\node_modules\angular-cli\models\webpack-build-typescript.js:40:13)
    at new NgCliWebpackConfig (C:\Projects\ng2-bootstrap-3\node_modules\angular-cli\models\webpack-config.js:22:42)
    at Class.run (C:\Projects\ng2-bootstrap-3\node_modules\angular-cli\tasks\serve-webpack.js:20:22)
    at C:\Projects\ng2-bootstrap-3\node_modules\angular-cli\commands\serve.js:102:26
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

Maybe this has something to do with it?

**Comment by valorkin** [comment]
> after running npm install on the clone.

of course you don't
you need to build it first
with `npm run build`

**Comment by valorkin** [comment]
there are `npm run link` to work with ng2-bootstrap locally
https://docs.npmjs.com/cli/link
INPORTANT you should do not `npm link` but `npm run link`

**Comment by quiringk** [comment]
I got it to work with the carousel changes I added to 1.1.16-9. Here is the note I left myself for the future:

If code needs to be added to ng2-bootstrap's latest build, clone it, run "npm run build" and "npm install". If you want to run it locally, run "npm run link" and then "npm start." vscode will ask you about launch settings. the last time this worked for me I saw in the demo/bs-config.json file the port of 4200. So I copied my launch.json from our project to ng2-bootstrap project and changed the port from 3000 to 4200, went to localhost:4200 and everything worked and I was able to debug inside of vscode. To add these changes to our project, run "npm pack" inside of the dist/ directory in ng2-bootstrap. Then run "npm install tarfile" inside of our project to install the changes.
